src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -136,6 +141,11 @@ internal WebApplicationBuilder(WebApplicationOptions options, bool slim, Action<
             {
                 AspNetCore.WebHost.UseKestrel(webHostBuilder);
 
+                webHostBuilder.ConfigureServices(services =>
+                {
+                    services.AddRoutingCore();
+                });
+
```

Comment:
Why do we need this call?

Reply:
It is completely redundant here, but also somewhat defensive. Really the call that ```UseKestrel(...)``` makes to ```AddRoutingCore(...)``` is a side effect of the way it is implemented. The method naming doesn't really suggest that it also adds routing. So I can remove this redundant call and save a few cycles or leave it here defensively if ```UseKestrel(...)``` ever evolves.

One of the things that I've been struggling with is the "zen" of the ```Add[insert service name here](...)``` calls. Should code that depends on them assume that another class calls it or not. In the case of the default builder it does, but not in the case of the slim builder.

Reply:
While I agree with that generally, it's fair to say that routing is a core service. That method should be renamed (feel free to do it as part of this PR).

Reply:
I've gone with ```ConfigureWebDefaultsCore``` to mirror the method that it was extracted from. Combined with ```AddRoutingCore``` it forms the basis for a bit of a pattern that we are seeing where we have critical elements, we need to configure in the slim builder scenario which can be reused in the non-native AOT scenarios.